### DISCUSSION TITLE
Logging multimap hits

### DISCUSSION THREAD

**Discussion by ManChicken1911** [rspamd/rspamd#4631] [Q&A] [answered]
Before I go digging through a bunch of source trying to figure it out (I don't speak much LUA) is there a hidden or undocumented option  or something else I'm maybe missing to make rspamd log multimap hits telling me the exact line (either the content or even just the line number) of the multimap that matched?

Once in awhile I will get a false-positive because a regex I wrote is too broad or happens to match something innocent, or whatever -- but at the moment all I get is my symbol showing which multimap(s) had the hits, but then I have to go through and try and figure out what particular pattern did it which is not always obvious.

**Comment by fatalbanana** [comment]
You could [set symbols for some/all of those regexes](https://rspamd.com/doc/modules/multimap.html#multiple-symbol-maps), or if you'd set `disable_multisymbol = true` in the multimap rule configuration then whatever values you've set in maps would be appended to symbol 'options' instead of treated as described.

It might suit to use regex rules rather than multimap:
https://rspamd.com/doc/tutorials/writing_rules.html
https://rspamd.com/doc/modules/regexp.html

**Comment by sts** [comment, marked as answer]
@ManChicken1911 I'm generating map files via a script and I'm adding a hash of the rule to each rule, since 3.6 you can add options to a multi map lookup:

```
/Subject:.*kaufen Sie ohne Rezept/ DE_PILLS:6:vcb42efb
/Subject:.*Generikas garantiert/ DE_PILLS:6:v74f77b7
```

This will add the option vcb42efb or v74f77b7 to the DE_PILLS symbols, and you are able to trace it down to the exact regex. See https://github.com/rspamd/rspamd/issues/4468

**Reply by ManChicken1911** [reply]
Finally got around to looking at this and it seems to work and takes care of my issue. Thanks for bringing the feature to my attention @sts - guess the documentation just needs to catch up :)

I only wish that the score could be left out and have the 'default' score apply as defined in the multimap config, so if I wanted to change the general score I don't have to modify every rule, but it's not something I've actually changed in a very long time so I can live with it.
